src/gui/symbology/qgsrulebasedrendererwidget.cpp

```diff
@@ -1260,7 +1260,7 @@ bool QgsRuleBasedRendererModel::dropMimeData( const QMimeData *data,
     QDomElement ruleElem = rootElem.firstChildElement( QStringLiteral( "rule" ) );
     if ( rootElem.attribute( QStringLiteral( "type" ) ) == QLatin1String( "labeling" ) )
       _labeling2rendererRules( ruleElem );
-    QgsRuleBasedRenderer::Rule *rule = QgsRuleBasedRenderer::Rule::create( ruleElem, symbolMap );
+    QgsRuleBasedRenderer::Rule *rule = QgsRuleBasedRenderer::Rule::create( ruleElem, symbolMap )->clone();
```

Comment:
This leaks, because ::create returns a new object which is never cleaned up. I'd suggest instead adding a `bool regenerateNewKey = false` argument to  QgsRuleBasedRenderer::Rule::create and setting it to true here instead.

Also note that the same changes will need to be made to QgsRuleBasedLabelingModel::dropMimeData